### PR TITLE
Update glib to version 2.74.1

### PR DESCRIPTION
**Pull request by danyeaw**
- Update pcre2 to use meson to resolve link errors
- Update glib to version 2.74.1
- Update glib-networking to version 2.74.0

### CODE REVIEW

**Comment by danyeaw** [comment]
Anyone have an idea how to resolve this?
```
2022-10-09T15:35:01.1377284Z [105/1139] Compiling C object glib/glib-2.0-0.dll.p/gthread-win32.c.obj
2022-10-09T15:35:01.1378137Z ../glib/gthread-win32.c(359): warning C4312: 'type cast': conversion from 'DWORD' to 'gpointer' of greater size
2022-10-09T15:35:01.8775982Z Traceback (most recent call last):
2022-10-09T15:35:01.8788682Z   File "D:\a\gvsbuild\gvsbuild\gvsbuild\utils\builder.py", line 531, in build
2022-10-09T15:35:01.8789214Z     if self.__build_one(p):
2022-10-09T15:35:01.8789685Z   File "D:\a\gvsbuild\gvsbuild\gvsbuild\utils\builder.py", line 667, in __build_one
2022-10-09T15:35:01.8790111Z     skip_deps = proj.build()
2022-10-09T15:35:01.8790543Z   File "D:\a\gvsbuild\gvsbuild\gvsbuild\projects\glib.py", line 46, in build
2022-10-09T15:35:01.8790923Z     Meson.build(self)
2022-10-09T15:35:01.8791356Z   File "D:\a\gvsbuild\gvsbuild\gvsbuild\utils\base_builders.py", line 56, in build
2022-10-09T15:35:01.8792004Z     self.builder.exec_ninja(params="install", working_dir=ninja_build)
2022-10-09T15:35:01.8792810Z   File "D:\a\gvsbuild\gvsbuild\gvsbuild\utils\builder.py", line 989, in exec_ninja
2022-10-09T15:35:01.8793300Z     self.__execute(
2022-10-09T15:35:01.8793846Z   File "D:\a\gvsbuild\gvsbuild\gvsbuild\utils\builder.py", line 1045, in __execute
2022-10-09T15:35:01.8794588Z     subprocess.check_call(args, cwd=working_dir, env=env, shell=True)
2022-10-09T15:35:01.8795065Z   File "C:\hostedtoolcache\windows\Python\3.10.7\x64\lib\subprocess.py", line 369, in check_call
2022-10-09T15:35:01.8795715Z     raise CalledProcessError(retcode, cmd)
2022-10-09T15:35:01.8796792Z subprocess.CalledProcessError: Command 'ninja install' returned non-zero exit status 1.
2022-10-09T15:35:01.8797525Z C:\gtk-build\build\x64\release\glib\glib\gthread-win32.c(463) : warning C4702: unreachable code
2022-10-09T15:35:01.8798412Z [106/1139] Compiling C object glib/gspawn-win64-helper.exe.p/gspawn-win32-helper.c.obj
2022-10-09T15:35:01.8799133Z ../glib/gspawn-win32-helper.c(334): warning C4456: declaration of 'i' hides previous local declaration
2022-10-09T15:35:01.8799846Z ../glib/gspawn-win32-helper.c(210): note: see declaration of 'i'
2022-10-09T15:35:01.8800257Z [107/1139] Linking target glib/glib-2.0-0.dll
2022-10-09T15:35:01.8800644Z FAILED: glib/glib-2.0-0.dll glib/glib-2.0-0.pdb 
2022-10-09T15:35:01.8801187Z "link" @glib/glib-2.0-0.dll.rsp
2022-10-09T15:35:01.8801573Z LINK : fatal error LNK1181: cannot open input file 'pcre2-8.lib'
```

**Comment by nacho** [comment]
I think we should wait to update glib, I recall @pbor telling me that there was a bug with gtk + glib 2.74 on windows

**Comment by pbor** [comment]
The issue I had in mind is https://gitlab.gnome.org/GNOME/gtk/-/issues/5207 which despite the title, is a gtk issue

**Comment by pbor** [comment]
I think that the compile problem is that glib now uses pcre2 instead of 1, so we need to add that

**Comment by danyeaw** [comment]
I did add pcre2

File: gvsbuild/projects/glib.py [hunk 1]
```diff
@@ -26,13 +26,11 @@ def __init__(self):
         Project.__init__(
             self,
             "glib",
-            archive_url="https://download.gnome.org/sources/glib/2.72/glib-2.72.4.tar.xz",
```
Comment (saste) [inline gvsbuild/projects/glib.py:29]: Can this be build locally?

latest glib was updated to use pcre2, previously it was downloading pcre when building glib. Now it is needed to add  a pcre2 project, then an explicit dependency from glib to pcre2.

**Comment by pbor** [comment]
> I did add pcre2

I do not see it in

```
dependencies=["ninja", "meson", "pkg-config", "gettext", "libffi", "zlib"],
```

**Comment by danyeaw** [comment]
I did add a pcre2 project. However I was getting the linker error. Meson is now automatically pulling in and building pcre2, and the meson build settings are working great

**Comment by pbor** [comment]
> I did add a pcre2 project. However I was getting the linker error. Meson is now automatically pulling in and building pcre2, and the meson build settings are working great

I am worried about taking the slippery slope of relying on meson dependency management: for instance when we generate source tarballs (which is an important for our open source compliance) would pcre2 be included?

**Comment by danyeaw** [comment]
I don't feel strongly about this since we are providing a means to compile GTK, we aren't packaging it. If meson is managing some of our dependencies for us, it is one less thing for us to manage.

If we do want to add the dependencies, we'll need to add gvdb in addition to pcre2. I noticed that the [glib meson.build](https://gitlab.gnome.org/GNOME/glib/-/blob/main/meson.build) has some static build stuff for Windows:

```
# Try again with the fallback
if not pcre2.found()
  pcre2 = dependency('libpcre2-8', required : true, fallback : ['pcre2', 'libpcre2_8'])
  use_pcre2_static_flag = true
elif host_system == 'windows'
  pcre2_static = cc.links('''#define PCRE2_STATIC
                             #define PCRE2_CODE_UNIT_WIDTH 8
                             #include <pcre2.h>
                             int main() {
                               void *p = NULL;
                               pcre2_code_free(p);
                               return 0;
                             }''',
                         dependencies: pcre2,
                         name : 'Windows system PCRE2 is a static build')
  use_pcre2_static_flag = pcre2_static
else
  use_pcre2_static_flag = false
endif
```

**Comment by pbor** [comment]
I still think I would prefer for us to manage it. I see gvsbuild as a way to create a bundle of dependencies and if later we have another lib that uses pcre2 I want to build it and ship it just once

**Comment by pbor** [comment]
Thanks for figuring this out :)

**Pull request merged**